### PR TITLE
Iteration number expression

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?223845

@sheelio does this solve your problem?

cc @sravfeyn @nickpell

### CODE REVIEW

File: corehq/apps/userreports/README.md [hunk 1]
```diff
@@ -829,7 +840,8 @@ All indicators output single values. Though fractional indicators are common, th
 ## Saving Multiple Rows per Case/Form
 
 You can save multiple rows per case/form by specifying a root level `base_item_expression` that describes how to get the repeat data from the main document.
-You can also use the `root_doc` expression type to reference parent properties.
+You can also use the `root_doc` expression type to reference parent properties
```
Comment (sravfeyn) [inline corehq/apps/userreports/README.md]: Sweet!

**Comment by sravfeyn** [comment]
LGTM, will wait for tests

**Comment by sheelio** [comment]
@czue 

Is it possible to get the current expression of the base_item_expression instead of just the index?  For this particular situation, I can make do with just the iteration number (since my expressions are just numbers), but would useful in the future to get the current expression.

**Comment by czue** [comment]
@sheelio like the expression json itself? or the compiled thing you can run? (i think we should move this in either way since that sounds quite a bit harder)

**Comment by sheelio** [comment]
@czue 

For example, my base item expression looks like this:
{
  "test": {}, 
  "expressions": [
    -3, 
    -2, 
    -1, 
    0, 
    1, 
    2, 
    3
  ], 
  "type": "iterator"
}

Typically, using the "identity" expression will return -2, -1, etc. depending on which iteration I'm in.  Is there a way to replicate this behavior?  It sounds like  "base_iteration_number " will just return 0, 1, etc.

**Comment by czue** [comment]
I think I need to better understand why you want to do that to answer the question. We can follow up on our chat today. In the meantime I think using this + eval should work to get those numbers otu?

**Comment by sheelio** [comment]
@czue 
Can definitely make this work using eval expression and just starting at index 0

Current Situation:
- My base_item_expressions are -3, -2, -1, 0, 1, 2, 3
- I calculate current_month using start_of_month( add_month(case_update_date, IDENTITY) )
- Assuming a case was updated on May 20, 2015, this would result in: 2015-02-01, 2015-03-01, 2015-04-01, 2015-05-01, 2015-06-01

I guess its just a bit odd that there isn't a way to replicate the behavior of IDENTIFY when I'm within a filter_items/map_items

**Comment by czue** [comment]
Hey - yeah I see what you mean. I think I need to think about it more to come up with something better. I think there's probably something clever that we can do with the evaluation context but it's a more complicated thing than I can commit to working on right now.

**Comment by sheelio** [comment]
@czue 

Okay - is fine to merge this change then so I can build out the basic datasources using eval expression?

**Comment by czue** [comment]
:+1: cc @sravfeyn @snopoke

**Pull request merged**